modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/provider/APIApplicationProviderImpl.java

```diff
@@ -316,7 +316,7 @@ private APIApplication _toApiApplication(
 
 			@Override
 			public String getBaseURL() {
-				return (String)properties.get("baseURL");
+				return "/c/" + (String)properties.get("baseURL");
```

Comment:
I would not put the `/c/` here. Instead I would use it only inside `APIApplicationPublisher`. The reason is that someone could create an APIApplication that does not returbn `/c/` and that could collide with existing apps. If we add it inside the Publisher, we will avoid this problem and every API Builder app will be deployed under `/c/` no matter what.

Reply:
Do you mean just hardcoding the `baseURL` in the `APIApplicationPublisher`?

Reply:
I just follow this pattern because it is very similar to Objects, as well as I don't know how could this collide because we are abstracting the implementation. If an user wants to create its API Application, you don't have to indicate the `/c/`. What do you think @4lejandrito ?

Reply:
The only code that should know about `/c/` I think it should be the `APIApplicationPublisher`. That way it's centralized and does not depend on the implementation of the `APIApplication`.